streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/ForeignJoinSubscriptionSendProcessorSupplier.java

```diff
@@ -217,5 +207,26 @@ public void process(final Record<K, Change<V>> record) {
                 }
             }
         }
+
+        @Override
+        public void close() {
+            primaryKeyValueGetter.close();
+        }
+
+        private void logSkippedRecordDueToNullForeignKey() {
```

Comment:
Extracting the common code from `process()` into this helper method is purely a refactor, unrelated to the changes in this PR. I had to make the refactor in order to avoid a CyclomaticComplexity checkstyle failure upon adding a clause into `process()` for dropping out-of-order records.